Beginning of new Sphinx tutorial (#9276)
sphinx-doc/sphinx: doc/usage/installation.rst

   $ pip install -U --pre sphinx

Using virtual environments
~~~~~~~~~~~~~~~~~~~~~~~~~~

When installing Sphinx using pip,
it is highly recommended to use *virtual environments*,

> which isolate the Installation

Reviewer: ```suggestion
which isolate the installed packages from the system packages,
```

> and remove the need to use administrator privileges.

Reviewer: ```suggestion
thus removing the need to use administrator privileges.
```

To create a virtual environment in the ``.venv`` directory,
use the following command.

::

>    $ python -m venv .venv

Reviewer: One more reason to use `env` instead of `.venv`, the PyPA uses this in its documentation:

`python3 -m venv env`

or

`py -m venv env`

Being consistent with the PyPA docs is a very Good Thing™.

Author: > Being consistent with the PyPA docs is a very Good Thing™.

I might be excessively nitpicky with this, but the Python Packaging Authority has stated several times that they are not an authority, despite the name :) Many people ([including myself](https://twitter.com/juanluisback/status/1398623726070022144)) have fallen this trap several times.

Author: Having said that:

1. All options seem to be somewhat used, judging at the output of https://www.gitignore.io/api/python:

```
# Environments
# .env
.env/
.venv/
env/
venv/
ENV/
env.bak/
venv.bak/
pythonenv*
```

2. Consistency with the PyPA docs is a good thing in itself, but [A Foolish Consistency is the Hobgoblin of Little Minds](https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds). I like `.venv` because it's hidden, and I avoid `env` or `.env` to remove confusion with [direnv](https://direnv.net/) and similar tools. However, I'm sure they have their own reasons to adopt `env`, and probably either name is just fine.
3. On the other hand, giving it a different name from the PyPA docs also communicates that it doesn't really matter, or that the user can change it if they want.
4. Like all serious things in life, [let's decide them with a Twitter poll](https://twitter.com/juanluisback/status/1399375962228224005) (I'm joking, but anyway I'm curious about the results)

Reviewer: 1. OK. I see `env` is common usage.
2. Would this consistency with the PyPA example be foolish?
3. That would be a sidebar to creating a virtual environment.
4. There's also strong opinions about where to put them, either inside the project folder or in the user home directory.

For tutorials, I strongly advocate for keeping things as simple as possible on all levels to avoid confusion. If you have ever conducted any training, you know how difficult it is to get people to install packages into a virtual environment in the first place. That's the background to my reasons for using `env` in my [original comment](https://github.com/sphinx-doc/sphinx/pull/9276#discussion_r641983256).

Reviewer 2: > `python3 -m venv env`

Only because I need to update https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/?highlight=venv#creating-a-virtual-environment. 😅 

In general, I think a lot of the material is converging toward `.venv` and the fact that packaging.python.org is lagging is due to lack of volunteer time to update it more than anything else.

> Being consistent with the PyPA docs is a very Good Thing™.

WELL. I guess it's a good thing that I'm currently working on improving PyPA documentation stuff then. ;)

Reviewer 2: > Python Packaging Authority has stated several times that they are not an authority, despite the name :)

As an aside, there is definitely authority associated with the PyPA now -- Python Steering Council officially delegates power for deciding on all Python Packaging things to the PyPA. I think we're aware of the "what we say matters" situation, but that is not as coordinated/structured as folks would expect from "authority". :)

Reviewer 2: All this is to say, `.venv` works and I don't think it needs to change to be in sync with packaging.python.org; although I'll likely update that to match this style later. :)

You can read more about them in the `Python Packaging User Guide`_.

.. _Python Packaging User Guide: https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment

.. warning::

>    Note like in some Linux distributions like Debian and Ubuntu

Reviewer: ```suggestion
   Note that in some Linux distributions, such as Debian and Ubuntu,
```

>    this might require an extra installation step::

Reviewer: ```suggestion
   this might require an extra installation step as follows.

    .. code-block:: bash
   
```

>      $ apt-get install python3-venv

Reviewer: Remove prompts as they are often copy-pastad and they are different across operating systems.
```suggestion
      apt-get install python3-venv
```

Author: I hear your point about the prompt being pasted from time to time, but

1. It's a good way of communicating that it's a Linux/UNIX-like command (unlike, say, Windows), and
2. It keeps consistency with other code blocks that retain the Bash output

What do you think?

Reviewer 3: I think you can use `.. code:: console` to avoid having those being copied.

Author: TIL: https://pygments.org/docs/lexers/?highlight=console#pygments.lexers.shell.BashSessionLexer

Thanks @stsewd ! Changing it everywhere in this PR now.
